src/Castle.Windsor/MicroKernel/ModelBuilder/Inspectors/LifestyleModelInspector.cs

```diff
@@ -193,9 +193,9 @@ protected virtual void ValidateTypeFromAttribute(Type typeFromAttribute, Type ex
 						scopeRootBinderType.Name, name));
 			}
 			var instance = scopeRootBinderType.CreateInstance<object>();
-			return
-				(Func<IHandler[], IHandler>)
-				Delegate.CreateDelegate(typeof(Func<IHandler[], IHandler>), instance, (MethodInfo)filterMethod);
+
+			MethodInfo methodInfo = filterMethod as MethodInfo;
```

Comment:
This shouldn't use a safe-cast, otherwise the following line will just throw a null-ref exception.